pkg/driver/distro/generic.go

```diff
@@ -68,16 +67,11 @@ func (g *generic) customizeBuild(_ context.Context,
 }
 
 //nolint:gocritic // the method shall not be able to modify kr
-func (g *generic) PreferredDriver(kr kernelrelease.KernelRelease) drivertype.DriverType {
-	// Deadly simple default automatic selection
-	var dType drivertype.DriverType
-	switch {
-	case kr.GTE(semver.MustParse("5.8.0")):
-		dType, _ = drivertype.Parse(drivertype.TypeModernBpf)
-	case kr.SupportsProbe():
-		dType, _ = drivertype.Parse(drivertype.TypeBpf)
-	default:
-		dType, _ = drivertype.Parse(drivertype.TypeKmod)
+func (g *generic) PreferredDriver(kr kernelrelease.KernelRelease, allowedDriverTypes []drivertype.DriverType) drivertype.DriverType {
```

Comment:
Also, i plan to reuse the `drivertype.Supported()` logic in https://github.com/falcosecurity/dbg-go, by adding a new option like: `--skip-modernbpf-supported` (that will initially default to false).
In this way, once we enable this new flag, we will start generating configs only for kernels where the modern bpf cannot run. 
Probably Falco 1.0.0 will be the right time to enable the flag.

cc @leogr wdyt?

Reply:
Here is the PR: https://github.com/falcosecurity/dbg-go/pull/43

Reply:
Uh that's wrong, since the PreferredDriver function for modern bpf directly checks running kernel, thus we cannot rely on it against any kernel release version.